lisa/messages.py

```diff
@@ -137,3 +127,54 @@ class NetworkPPSPerformanceMessage(PerfMessage):
     rx_tx_pps_minimum: Decimal = Decimal(0)
     rx_tx_pps_average: Decimal = Decimal(0)
     rx_tx_pps_maximum: Decimal = Decimal(0)
+
+
+@dataclass
+class NetworkNtttcpTCPPerformanceMessage(PerfMessage):
+    connections_num: int = 0
+    throughput_in_gbps: Decimal = Decimal(0)
+    latency_us: Decimal = Decimal(0)
+    buffer_size: Decimal = Decimal(0)
+    tx_packets: Decimal = Decimal(0)
+    rx_packets: Decimal = Decimal(0)
+    pkts_interrupts: Decimal = Decimal(0)
+    number_of_receivers: int = 1
+    number_of_senders: int = 1
+    sender_cycles_per_byte: Decimal = Decimal(0)
+    connections_created_time: int = 0
+    retrans_segments: int = 0
+    receiver_cycles_rer_byte: Decimal = Decimal(0)
+
+
+@dataclass
+class NetworkNtttcpUDPPerformanceMessage(PerfMessage):
+    connections_num: int = 0
+    number_of_receivers: int = 1
+    number_of_senders: int = 1
+    connections_created_time: int = 0
+    receiver_cycles_rer_byte: Decimal = Decimal(0)
+    send_buffer_size: Decimal = Decimal(0)
+    tx_throughput_in_gbps: Decimal = Decimal(0)
+    rx_throughput_in_gbps: Decimal = Decimal(0)
+    data_loss: Decimal = Decimal(0)
+    packet_size_kbytes: Decimal = Decimal(0)
+
+
+@dataclass
+class NetworkIperfSingleTCPPerformanceMessage(PerfMessage):
+    buffer_size_bytes: Decimal = Decimal(0)
+    tx_throughput_in_gbps: Decimal = Decimal(0)
+    rx_throughput_in_gbps: Decimal = Decimal(0)
+    retransmitted_segments: Decimal = Decimal(0)
+    congestion_windowsize_kb: Decimal = Decimal(0)
+
+
+@dataclass
+class NetworkIperfUDPPerformanceMessage(NetworkNtttcpUDPPerformanceMessage):
+    pass
+
+
+def create_message_list(messages: List[T], information: Dict[str, str]) -> List[T]:
```

Comment:
The key things is to create the message objects in this method, if the message is created outside of the method, it's not useful. It doesn't need to create multiple messages. Try like below,

```python
create_message(message_type: Type[T], node:Node, test_case_name:str="", other_fields: Optional[Dict[str, Any]]=None) -> T:
    message = message_type()
   # fill in other fields from the node
    dict_to_fields(information, message)
    message.test_case_name = test_case_name

    if other_fields:
        dict_to_fields(other_fields, message)
    return message
```

And it's used like below,
```python
        for result in mode_iops_latency.values():
            fio_result_message = create_message(DiskPerformanceMessage, node, test_name, results)
            fio_message.append(fio_result_message)
        return fio_message
```